support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+// ----- Imports ----- //
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import { fetchJson } from 'helpers/fetch';
+import { logException } from '../logger';
+import { DirectDebit, ExistingCard, ExistingDirectDebit, Stripe } from '../paymentMethods';
+import { getPaymentLabel } from '../checkouts';
+
+// ----- Types ----- //
+
+export type ExistingPaymentMethodSubscription = {
+  isActive: boolean;
+  isCancelled: boolean,
+  name: string
+}
+
+type ExistingPaymentType = 'Card' | 'DirectDebit';
+
+export type PartialExistingPaymentMethod = {
+  paymentType: ExistingPaymentType;
+};
+
+export type FullDetailExistingPaymentMethod = {
+  billingAccountId: string,
+  subscriptions: ExistingPaymentMethodSubscription[],
+  card?: string,
+  mandate?: string,
+  // can't use Intersection types to avoid repeating this as it breaks isFullDetailExistingPaymentMethod because; Flow!
+  paymentType: ExistingPaymentType
```

Comment:
you can use the spread operator with Flow types as well - that might work for this use case?

Reply:
tried this but Flow still complains elsewhere, for example... 
![image](https://user-images.githubusercontent.com/19289579/55171419-3a6e3980-5170-11e9-9837-ebba321c99f8.png)